### ISSUE TITLE
NVDA not reading the menu items properly in Dev C++

### ISSUE THREAD

**Issue by Bibeksharma5** [nvaccess/nvda#20158]
### Brief summary

Hello NVAccess team, I want to report you one bug of NVDA 2026.1 while working in Dev C++. I was also experiencing this problem with balabolka. NVDA is not readin the menu items properly in dev C++. Most of the time, it doesn't read the lable of the menus. It says "menu" when I press up/down arrow keys. If it reads, it doesn't read the menu items. It reads just like mixup of text. This is a new bug. While using NVDA 2025.3.3, this problem was not with NVDA. NVDA was reading them fine. But, NVDA 2026.1 is not doing that. Just now, I tested with NVDA 2025.3.3. Everything is fine. So, please fix this bug. Regards.

### Steps to reproduce

1. Open Dev c++.
2. Press alt key to go through menus.
3. Press up/down and left/right to experience this issue.
4. Click on new>source file.
5. Again, press alt key to go through menu bar.
6. Repeat step 3.

### Actual behavior

NVDA not reading the menu items and if it does, it is not properly reading.
NVDA speech viewer with "speak command keys" enabled:
"NVDA Speech Viewer
Show Speech Viewer on Startup  check box  not checked  Alt+  s
speak command keys on
windows+m
Desktop  list
Dev-C++  2 of 24
enter
Dev-C++ 5.11
tbCompilers  tool bar
combo box  TDM-GCC 4.9.2 64-bit Release  collapsed
File  subMenu  Alt+  f
down arrow
New  subMenu
down arrow
New
up arrow
New  subMenu
right arrow
Source File
down arrow
menu item
up arrow
Source File
enter
Untitled1 - Dev-C++ 5.11
tab control
Untitled1  property page
edit    1

File  subMenu  Alt+  f
down arrow
New  subMenu
down arrow
New
down arrow
Open... Ct  unavailable
down arrow
Open... CtSave C
down arrow
Save CSave As...  unavailable
down arrow
Save As...Save Project As...  unavailable
down arrow
Save Project As...Save All Shift+C
down arrow
Save All Shift+C  unavailable
down arrow
Close Ct
down arrow
Close Project
down arrow
Close All Shift+Ct  subMenu
down arrow
Close All Shift+CtProperties...  subMenu
down arrow
Properties...  unavailable
down arrow
Import
down arrow
Export
down arrow
New  subMenu
right arrow
Source File
down arrow
menu item
down arrow
menu item  unavailable
down arrow
menu item  unavailable
escape
New  subMenu
escape
File  subMenu  Alt+  f
right arrow
Edit  subMenu  Alt+  e
down arrow
Undo  unavailable
down arrow
menu item  unavailable
down arrow
menu item  unavailable
down arrow
menu item  unavailable
down arrow
menu item
down arrow
menu item  unavailable
down arrow
menu item  subMenu
down arrow
menu item  subMenu
down arrow
menu item  subMenu
down arrow
menu item  unavailable
down arrow
menu item  unavailable
down arrow
menu item  unavailable
down arrow
menu item  unavailable
down arrow
menu item
down arrow
menu item
escape
Edit  subMenu  Alt+  e
right arrow
Search  subMenu  Alt+  s
down arrow
Find...
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item  unavailable
escape
Search  subMenu  Alt+  s
right arrow
View  subMenu  Alt+  v
down arrow
Project/Class Browser  checked
down arrow
menu item  checked
down arrow
menu item  subMenu
down arrow
menu item
down arrow
menu item  unavailable
down arrow
menu item
down arrow
menu item
down arrow
menu item
escape
View  subMenu  Alt+  v
right arrow
Project  subMenu  Alt+  p
down arrow
New File  unavailable
down arrow
menu item  unavailable
down arrow
menu item  unavailable
escape
Project  subMenu  Alt+  p
right arrow
Execute  subMenu  Alt+  x
down arrow
Compile
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item
escape
Execute  subMenu  Alt+  x
right arrow
Tools  subMenu  Alt+  t
down arrow
Compiler O
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item
down arrow
menu item
escape
Tools  subMenu  Alt+  t
right arrow
AStyle  subMenu  Alt+  a
right arrow
Window  subMenu  Alt+  w
left arrow
AStyle  subMenu  Alt+  a
down arrow
Format Current Fi  unavailable
down arrow
menu item
down arrow
Format Current Fi  unavailable
down arrow
menu item
escape
AStyle  subMenu  Alt+  a
right arrow
Window  subMenu  Alt+  w
right arrow
Help  subMenu  Alt+  h
down arrow
Hel
down arrow
menu item
down arrow
menu item
escape
Help  subMenu  Alt+  h
escape
Untitled1 - Dev-C++ 5.11
tab control
Untitled1  property page
edit    1

alt+f4
NVDA Speech Viewer
Show Speech Viewer on Startup  check box  not checked  Alt+  s
tab"

### Expected behavior

NBDA should properly read all the menu items.

### NVDA logs, crash dumps and other attachments

[nvda.log](https://github.com/user-attachments/files/27939232/nvda.log)
[nvda-old.log](https://github.com/user-attachments/files/27939233/nvda-old.log)

### NVDA type

installed copy

### NVDA version

2026.1 (2026.1.0.55743)

### Have you tried any other versions of NVDA? If so, please report their behaviors.

Yes, I've tried with NVDA 2025.3.3. This bug is not there.

### Windows version

Windows 11 25H2 (AMD64) build 26200.8457

### Name and version of other software in use when reproducing the issue

Version 2, June 1991

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the System Accessibility Repair Tool (COM Registration Fixing Tool in older versions) in NVDA's tools menu?

I have run the System Accessibility Repair Tool and the issue still occurs

**Comment by seanbudd** [comment]
Can you clarify what version of Dev C++ you are using? The issue template says "Version 2, June 1991" but the initial release was 1998 and the latest version is 6.3. Are you really on version 2? Why not 6.3?

**Comment by Bibeksharma5** [comment]
I don't know. I saw this in the about dialog of dev c++.

**Comment by seanbudd** [comment]
I believe you copied the license information. Can you make sure to find out the version information somehow?

**Comment by Bibeksharma5** [comment]
When I launch dev c++, NVDA says "Dev c++ 5.11". This may be the version if.

**Comment by seanbudd** [comment]
can you try updating to dev C++ 6.3?

**Comment by Bibeksharma5** [comment]
Hi! The latest version is 5.11. I just checked in the SourceForge page.

**Comment by seanbudd** [comment]
You can find version 6.3 here: https://github.com/Embarcadero/Dev-Cpp/releases/tag/v6.3
The project was forked so this is from different authors, and this fork was also abandoned.

**Comment by Bibeksharma5** [comment]
Same result.

**Comment by josephsl** [comment]
Hi,

Based on the latest comment, it looks like this issue is a duplicate of #19225 and the display scaling workaround may apply here too. Therefore, I think we should get this fix into 2026.2 or 2026.3 as this is a regression from 64-bit transition/display model/32-bit app support.

Thakns.

**Comment by josephsl** [comment]
Hi,

As a follow-up, just to make sure; is the version of Dev C++ in use a 32-bit program? If yes, then this is indeed a duplicate of the mentioned issue.

Thanks.
